# Tarea 02: Manipulación y Procesamiento de Datos con Python
**Alumno:** Jesús Abel Vera Cruz
**Fecha:** 13 de julio de 2026
**Dataset:** train.csv  (misma carpeta para que pueda correr)

In [14]:
!pip install matplotlib
!pip install matplotlib seaborn

In [48]:
# Importación de librerías necesarias
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Configurar pandas para mostrar texto completo sin truncar
pd.set_option('display.max_colwidth', 60)   # Ancho máximo de columna en caracteres
pd.set_option('display.width', 120)         # Ancho total de la tabla en caracteres

# Guardar el CSV en la misma carpeta para que el evaluador pueda correrlo

# Cargar dataset (misma carpeta para que pueda correr)
df = pd.read_csv('train.csv')
# Mostrar información general del dataset
print("="*60)
print("Información general del dataset")
print("="*60)
print("Forma (filas, columnas)")
print("\n" + "="*60)
print("Tipos de datos por columna")
print("="*60)
print(df.dtypes)
print("\n" + "="*60)
print("Información detallada")
print("="*60)
df.info()
print("\n" + "="*60)
print("Variables nuemericas y categoricas")
print("="*60)
num_cols = df.select_dtypes(include=['int64', 'float64']).columns
cat_cols = df.select_dtypes(include=['string']).columns
print(f"Variables numéricas: {len(num_cols)}")
print(f"Variables categóricas: {len(cat_cols)}")
print(f"Total: {len(num_cols) + len(cat_cols)}")



Información general del dataset
Forma (filas, columnas)

Tipos de datos por columna
Id                 int64
MSSubClass         int64
MSZoning             str
LotFrontage      float64
LotArea            int64
                  ...   
MoSold             int64
YrSold             int64
SaleType             str
SaleCondition        str
SalePrice          int64
Length: 81, dtype: object

Información detallada
<class 'pandas.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 81 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             1460 non-null   int64  
 1   MSSubClass     1460 non-null   int64  
 2   MSZoning       1460 non-null   str    
 3   LotFrontage    1201 non-null   float64
 4   LotArea        1460 non-null   int64  
 5   Street         1460 non-null   str    
 6   Alley          91 non-null     str    
 7   LotShape       1460 non-null   str    
 8   LandContour    1460 non-null   str    
 9   Utiliti

In [30]:
# Diccionario con la descripción de todas las variables del dataset Ames Housing
variable_description = {
    # Variable objetivo
    'SalePrice': 'Precio de venta de la propiedad en dólares (variable objetivo)',
    
    # Características generales
    'MSSubClass': 'Clase de construcción',
    'MSZoning': 'Clasificación general de zonificación',
    'LotFrontage': 'Pies lineales de calle conectados a la propiedad',
    'LotArea': 'Tamaño del lote en pies cuadrados',
    'Street': 'Tipo de acceso vial',
    'Alley': 'Tipo de acceso a callejón',
    'LotShape': 'Forma general de la propiedad',
    'LandContour': 'Planitud de la propiedad',
    'Utilities': 'Tipo de servicios públicos disponibles',
    'LotConfig': 'Configuración del lote',
    'LandSlope': 'Pendiente de la propiedad',
    'Neighborhood': 'Ubicaciones físicas dentro de los límites de la ciudad de Ames',
    'Condition1': 'Proximidad a carretera principal o ferrocarril',
    'Condition2': 'Proximidad a carretera principal o ferrocarril (si hay un segundo)',
    'BldgType': 'Tipo de vivienda',
    'HouseStyle': 'Estilo de vivienda',
    'OverallQual': 'Calidad general de materiales y acabados',
    'OverallCond': 'Calificación general de condición',
    'YearBuilt': 'Fecha de construcción original',
    'YearRemodAdd': 'Fecha de remodelación',
    'RoofStyle': 'Tipo de techo',
    'RoofMatl': 'Material del techo',
    'Exterior1st': 'Cubierta exterior de la casa',
    'Exterior2nd': 'Cubierta exterior de la casa (si hay más de un material)',
    'MasVnrType': 'Tipo de revestimiento de mampostería',
    'MasVnrArea': 'Área de revestimiento de mampostería en pies cuadrados',
    'ExterQual': 'Calidad del material exterior',
    'ExterCond': 'Condición actual del material en el exterior',
    'Foundation': 'Tipo de cimientos',
    
    # Características del sótano
    'BsmtQual': 'Altura del sótano',
    'BsmtCond': 'Condición general del sótano',
    'BsmtExposure': 'Paredes de sótano a nivel de jardín o salida',
    'BsmtFinType1': 'Calidad del área terminada del sótano',
    'BsmtFinSF1': 'Pies cuadrados terminados de tipo 1',
    'BsmtFinType2': 'Calidad de la segunda área terminada (si está presente)',
    'BsmtFinSF2': 'Pies cuadrados terminados de tipo 2',
    'BsmtUnfSF': 'Pies cuadrados sin terminar del área del sótano',
    'TotalBsmtSF': 'Pies cuadrados totales del área del sótano',
    'BsmtFullBath': 'Baños completos en el sótano',
    'BsmtHalfBath': 'Medios baños en el sótano',
    
    # Características de la vivienda
    'Heating': 'Tipo de calefacción',
    'HeatingQC': 'Calidad y condición de la calefacción',
    'CentralAir': 'Aire acondicionado central',
    'Electrical': 'Sistema eléctrico',
    '1stFlrSF': 'Pies cuadrados del primer piso',
    '2ndFlrSF': 'Pies cuadrados del segundo piso',
    'LowQualFinSF': 'Pies cuadrados terminados de baja calidad (todos los pisos)',
    'GrLivArea': 'Pies cuadrados de área habitable por encima del nivel del suelo',
    'FullBath': 'Baños completos por encima del nivel del suelo',
    'HalfBath': 'Medios baños por encima del nivel del suelo',
    'BedroomAbvGr': 'Número de dormitorios por encima del nivel del sótano',
    'Kitchen': 'Número de cocinas',
    'KitchenQual': 'Calidad de la cocina',
    'TotRmsAbvGrd': 'Total de habitaciones por encima del nivel del suelo (no incluye baños)',
    'Functional': 'Calificación de funcionalidad del hogar',
    'Fireplaces': 'Número de chimeneas',
    'FireplaceQu': 'Calidad de la chimenea',
    
    # Características del garaje
    'GarageType': 'Ubicación del garaje',
    'GarageYrBlt': 'Año en que se construyó el garaje',
    'GarageFinish': 'Acabado interior del garaje',
    'GarageCars': 'Capacidad del garaje en autos',
    'GarageArea': 'Tamaño del garaje en pies cuadrados',
    'GarageQual': 'Calidad del garaje',
    'GarageCond': 'Condición del garaje',
    
    # Características exteriores
    'PavedDrive': 'Entrada pavimentada',
    'WoodDeckSF': 'Área de terraza de madera en pies cuadrados',
    'OpenPorchSF': 'Área de porche abierto en pies cuadrados',
    'EnclosedPorch': 'Área de porche cerrado en pies cuadrados',
    '3SsnPorch': 'Área de porche de tres estaciones en pies cuadrados',
    'ScreenPorch': 'Área de porche con mosquitero en pies cuadrados',
    'PoolArea': 'Área de la piscina en pies cuadrados',
    'PoolQC': 'Calidad de la piscina',
    'Fence': 'Calidad de la cerca',
    'MiscFeature': 'Característica miscelánea no cubierta en otras categorías',
    'MiscVal': 'Valor en dólares de la característica miscelánea',
    
    # Características de venta
    'MoSold': 'Mes de venta',
    'YrSold': 'Año de venta',
    'SaleType': 'Tipo de venta',
    'SaleCondition': 'Condición de la venta'
}
desc_df = pd.DataFrame(list(variable_description.items()), columns=['Variable', 'Descripción'])
# Calcular porcentaje de nulos para cada variable
missing_pct = df.isnull().mean() * 100
desc_df['% Nulos'] = desc_df['Variable'].map(missing_pct)

# Identificar si es numérica o categórica
num_cols = df.select_dtypes(include=['int64', 'float64']).columns
desc_df['Tipo'] = desc_df['Variable'].apply(lambda x: 'Numérica' if x in num_cols else 'Categórica')

# Mostrar las primeras filas para verificar
print("Diccionario de variables cargado exitosamente.")
print(f"Total de variables documentadas: {len(desc_df)}")
desc_df.head(10)

Diccionario de variables cargado exitosamente.
Total de variables documentadas: 80


,Variable,Descripción,% Nulos,Tipo
0,SalePrice,Precio de venta de la propiedad en dólares (va...,0.000000,Numérica
1,MSSubClass,Clase de construcción,0.000000,Numérica
2,MSZoning,Clasificación general de zonificación,0.000000,Categórica
3,LotFrontage,Pies lineales de calle conectados a la propiedad,17.739726,Numérica
4,LotArea,Tamaño del lote en pies cuadrados,0.000000,Numérica
5,Street,Tipo de acceso vial,0.000000,Categórica
6,Alley,Tipo de acceso a callejón,93.767123,Categórica
7,LotShape,Forma general de la propiedad,0.000000,Categórica
8,LandContour,Planitud de la propiedad,0.000000,Categórica
9,Utilities,Tipo de servicios públicos disponibles,0.000000,Categórica


### Pregunta 1: Análisis de valores faltantes

 Calcule
el
porcentaje
de
valores
faltantes
por
columna
con
df.isnull().mean().sort_values(ascending=False). Liste las 10 variables con
mayor porcentaje de valores nulos y clasifíquelas en: eliminar, imputar
simple o imputar avanzado, justificando el criterio usado (porcentaje
faltante y tipo de variable)


In [50]:
#1 calcular el porcentaje de valores faltantes
porcentajeValoresFaltantes=df.isnull().mean().sort_values(ascending=False)*100
df_porcentaje = pd.DataFrame({
    'Variable': porcentajeValoresFaltantes.index,
    '% Nulos': porcentajeValoresFaltantes.values,
    'Tipo':porcentajeValoresFaltantes.dtypes
})
df_porcentaje['Descripción'] = df_porcentaje['Variable'].map(variable_description)
print("="*260)
print("Porcentaje de valores faltantes los 10 más nulos:")
print(df_porcentaje.head(10))
print("\n" + "="*260)

Porcentaje de valores faltantes los 10 más nulos:
       Variable    % Nulos     Tipo                                                Descripción
0        PoolQC  99.520548  float64                                      Calidad de la piscina
1   MiscFeature  96.301370  float64  Característica miscelánea no cubierta en otras categorías
2         Alley  93.767123  float64                                  Tipo de acceso a callejón
3         Fence  80.753425  float64                                        Calidad de la cerca
4    MasVnrType  59.726027  float64                       Tipo de revestimiento de mampostería
5   FireplaceQu  47.260274  float64                                     Calidad de la chimenea
6   LotFrontage  17.739726  float64           Pies lineales de calle conectados a la propiedad
7    GarageQual   5.547945  float64                                         Calidad del garaje
8  GarageFinish   5.547945  float64                                Acabado interior del garaje


**Clasificación de variables con valores faltantes:**

- **Eliminar (nulos > 50%):** PoolQC (99.5%), MiscFeature (96.3%), Alley (93.8%), Fence (80.8%), MasVnrType (59.7%). No aportan información suficiente.
- **Imputación simple (nulos < 20%):** LotFrontage (17.7%), GarageQual (5.5%), GarageFinish (5.5%), GarageType (5.5%). Se imputarán con mediana/moda.
- **Imputación avanzada (nulos intermedios):** FireplaceQu (47.3%). Se imputará considerando su relación con Fireplaces.